package/libnetwork/src/libnetwork.c

```diff
@@ -169,9 +345,20 @@ static void network_request(CURL *curl)
   curl_easy_setopt(curl, CURLOPT_TCP_KEEPINTVL,     5L);
   curl_easy_setopt(curl, CURLOPT_TCP_KEEPIDLE,      20L);
   curl_easy_setopt(curl, CURLOPT_SOCKOPTFUNCTION,   network_sockopt);
```

Comment:
I'd move these two setopt's outside of this vs. passing in the pointer to transfer. This was just supposed to be a common request setup, where all of the data was setup outside of this function. Plus now that the transfer structure is capturing the socket fd, seems reasonable to keep it together with the other usages of the transfer pointer.